src/api/server.tsx

```diff
@@ -36,7 +37,30 @@ export const fetchResources = (): Promise<LxdResources> => {
   });
 };
 
-export const fetchConfigOptions = (): Promise<LxdConfigOptions> => {
+export const fetchConfigOptions: QueryFunction<LxdConfigOptions> = ({
+  queryKey,
+}) => {
+  const [, metadataConfigurationSupported] = queryKey;
```

Comment:
having a `useQuery` call in here simply doesn't work, I think it's because this is the fetch function for another `useQuery` call. Instead of having `useSupportedFeatures` called here, I moved it one level above.